allensdk/internal/api/behavior_ophys_api.py

```diff
@@ -87,21 +88,36 @@ def get_ophys_timestamps(self):
 
         ophys_timestamps = self.get_sync_data()['ophys_frames']
         dff_traces = self.get_raw_dff_data()
+        plane_group = self.get_imaging_plane_group()
+
         number_of_cells, number_of_dff_frames = dff_traces.shape
         num_of_timestamps = len(ophys_timestamps)
-        if number_of_dff_frames < num_of_timestamps:
-            ophys_timestamps = ophys_timestamps[:number_of_dff_frames]
-        elif number_of_dff_frames == num_of_timestamps:
-            pass
-        else:
-            raise RuntimeError('dff_frames is longer than timestamps')
-
+        # Scientifica data has extra frames in the sync file relative
+        # to the number of frames in the video. These sentinel frames
+        # should be removed.
+        # NOTE: This fix does not apply to mesoscope data.
+        # See http://confluence.corp.alleninstitute.org/x/9DVnAg
+        if plane_group is None:    # non-mesoscope
+            if (number_of_dff_frames < num_of_timestamps):
+                self.logger.info(
+                    "Truncating acquisition frames ('ophys_frames') to the "
+                    "number of frames in the df/f trace.")
+                ophys_timestamps = ophys_timestamps[:number_of_dff_frames]
+            elif number_of_dff_frames > num_of_timestamps:
+                raise RuntimeError('dff_frames is longer than timestamps')
```

Comment:
Might be good to get explicit counts for `number_of_dff_frames` and `num_of_timestamps` if they do not match.

Reply:
I believe you're looking at an outdated version :) I updated this already when I was looking back at the old code.